sap-vcap-services/src/test/java/com/sap/cloud/environment/servicebinding/SapVcapServicesServiceBindingAccessorTest.java

```diff
@@ -195,24 +195,42 @@ void resultIsNotCached()
     @Test
     void resultIsNotCachedWhenFileBased()
     {
-        final String vcapServicesFilePath =
+        final String vcapServicesFile =
             TestResource.getPathAsString(SapVcapServicesServiceBindingAccessorTest.class, "FullVcapServices.json");
+        final Path vcapServicesFilePath =
+            TestResource.get(SapVcapServicesServiceBindingAccessorTest.class, "FullVcapServices.json");
+        final String vcapServices =
+            TestResource.read(SapVcapServicesServiceBindingAccessorTest.class, "FullVcapServices.json");
         final CountingEnvironmentVariableReader environmentVariableReader = new CountingEnvironmentVariableReader();
         environmentVariableReader.addExpectedKeyAndValue("VCAP_SERVICES", null);
-        environmentVariableReader.addExpectedKeyAndValue("VCAP_SERVICES_FILE_PATH", vcapServicesFilePath);
+        environmentVariableReader.addExpectedKeyAndValue("VCAP_SERVICES_FILE_PATH", vcapServicesFile);
+        final CountingFileReader fileReader = new CountingFileReader(vcapServicesFilePath, vcapServices);
 
         final SapVcapServicesServiceBindingAccessor sut =
-            new SapVcapServicesServiceBindingAccessor(environmentVariableReader);
+            new SapVcapServicesServiceBindingAccessor(environmentVariableReader, fileReader);
 
         // first invocation
         assertThat(sut.getServiceBindings().size()).isEqualTo(3);
-        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES")).isEqualTo(1);
-        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES_FILE_PATH")).isEqualTo(1);
+        assertThat(fileReader.getInvocations()).isEqualTo(1);
 
         // second invocation
         assertThat(sut.getServiceBindings().size()).isEqualTo(3);
-        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES")).isEqualTo(2);
-        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES_FILE_PATH")).isEqualTo(2);
+        assertThat(fileReader.getInvocations()).isEqualTo(2);
+    }
+
+    @Test
+    void fileReaderThrowsIOException()
```

Comment:
maybe 
```suggestion
    void ioExceptionIsHandledGracefully()
```
is a bit clearer